src/getFlagsProxy.ts

```diff
@@ -60,15 +60,18 @@ function toFlagsProxy(ldClient: LDClient, flags: LDFlagSet, flagKeyMap: LDFlagKe
       const currentValue = Reflect.get(target, prop, receiver);
 
       // only process flag keys and ignore symbols and native Object functions
-      if (typeof prop === 'symbol' || !hasFlag(flagKeyMap, prop)) {
+      if (typeof prop === 'symbol' || (!hasFlag(flags, prop) && !hasFlag(flagKeyMap, prop))) {
         return currentValue;
       }
```

Comment:
We should pass `useCamelCaseFlagKeys` to the `toFlagsProxy` function so we can be more explicit with this flag existence check:

```suggestion
      // check if flag key exists as camelCase or original case
      const validFlagKey = (useCamelCaseFlagKeys && hasFlag(flagKeyMap, prop as string)) || hasFlag(target, prop as string);

      // only process flag keys and ignore symbols and native Object functions
      if (typeof prop === 'symbol' || !validFlagKey) {
        return currentValue;
      }
```